# **Tarea de Python - Ejercicio 3 - Visualización de Datos**

Se recolectaron datos de 8 estudiantes universitarios y se resumieron en la siguiente tabla:

<br>

<div align="center">

| MASA | ESTATURA | GENERO | ESTRATO | FUMA |
|:----:|:--------:|:------:|:-------:|:----:|
| 71   | 183      | HOMBRE | 2       | NO   |
| 45   | 168      | HOMBRE | 1       | NO   |
| 50   | 164      | HOMBRE | 1       | NO   |
| 60   | 164      | MUJER  | 3       | SI   |
| 61   | 167      | HOMBRE | 3       | SI   |
| 84   | 182      | HOMBRE | 3       | NO   |
| 55   | 168      | MUJER  | 3       | NO   |
| 60   | 170      | HOMBRE | 3       | SI   |

</div>

<br>

1. Crear un data frame y llamarlo __df__.

2. Realizar un análisis descriptivo básico de todas las variables.

3. Extraer un subconjunto de datos con los primeros 4 estudiantes y llamarlo __subc1__.

4. Extraer las variables categóricas y guardarlas en un subconjunto llamado __subc2__

5. Extraer las variables numéricas y guardarlas en un subconjunto llamado __subc3__

## **Desarrollo**

1. Primero, vamos a crear un data frame y lo llamaremos `df`:

In [1]:
import pandas as pd

# Crear el DataFrame
df = pd.DataFrame({
    'Masa': [71, 45, 50, 60, 61, 84, 55, 60],
    'Estatura': [183, 168, 164, 164, 167, 182, 168, 170],
    'Genero': ['HOMBRE', 'HOMBRE', 'HOMBRE', 'MUJER', 'HOMBRE', 'HOMBRE', 'MUJER', 'HOMBRE'],
    'Estrato': [2, 1, 1, 3, 3, 3, 3, 3],
    'Fuma': ['NO', 'NO', 'NO', 'SI', 'SI', 'NO', 'NO', 'SI']
})


df

Masa  Estatura  Genero  Estrato Fuma
0    71       183  HOMBRE        2   NO
1    45       168  HOMBRE        1   NO
2    50       164  HOMBRE        1   NO
3    60       164   MUJER        3   SI
4    61       167  HOMBRE        3   SI
5    84       182  HOMBRE        3   NO
6    55       168   MUJER        3   NO
7    60       170  HOMBRE        3   SI

2. Nos piden hacer un análasis descriptivo básico de todas las variables, primero empecemos con las variables numéricas, es decir, con `Masa`, `Estatura`, y `Fuma`.

Primero inicemos con la variable `Masa`:

In [2]:
result = pd.DataFrame({
    'Promedio': df[["Masa", "Estatura", "Estrato"]].mean(),
    'Desv. Estandar': df[["Masa", "Estatura", "Estrato"]].std(),
    'minimo': df[["Masa", "Estatura", "Estrato"]].min(),
    'maximo': df[["Masa", "Estatura", "Estrato"]].max(),
    'mediana': df[["Masa", "Estatura", "Estrato"]].median(),
    'Cuantil 1': df[["Masa", "Estatura", "Estrato"]].quantile(0.25),
    'Cuantil 3': df[["Masa", "Estatura", "Estrato"]].quantile(0.75),
    'IQR': df[["Masa", "Estatura", "Estrato"]].apply(lambda x: x.quantile(0.75) - x.quantile(0.25))
})

result.T

Masa   Estatura   Estrato
Promedio        60.750000  170.75000  2.375000
Desv. Estandar  12.209481    7.53563  0.916125
minimo          45.000000  164.00000  1.000000
maximo          84.000000  183.00000  3.000000
mediana         60.000000  168.00000  3.000000
Cuantil 1       53.750000  166.25000  1.750000
Cuantil 3       63.500000  173.00000  3.000000
IQR              9.750000    6.75000  1.250000

De acuerdo a los resultados obtenidos podemos destacar:

- El promedio de masa es de 60.75 kilogramos con una desviación estándar de 12.20 kilogramas. La mediana muestra que el 50% de las personas tiene una masa menor o igual a 60 kilogramas. El rango intercuartílico es de 9.75 kg lo que indica que el 50% central de los datos se encuentra entre 53.75 y 63.50 kilogramos. Por otro lado, el 25% de las personas tiene una masa menor o igual a 53.75 kilogramos, mientras que el 75% una menor o igual a 63.50 kilogramos. La masa mínima observada fue de 45 kilogramos y la máxima de 84 kilogramos.

- El promedio de estatura es de 170.75 centímetros con una desviación estándar de 7.53 cm. La estatura del 50% de las personas es menor o igual a 168 centímetros con un rango intercuartílico de 6.75, dandonos a entender que el 50% central de los datos se encuentra entre 166.25 y 173 centímetros. Por otro lado, el 25% de las personas tiene una estatura menor o igual a 166.25 centímetros y el 75% una menor o igual a 173 centímetros. La estatura mínima registrada fue de 164 cm y la máxima de 183 cm.

- El promedio de estrato es de 2.37 con una desviación estándar de 0.91. El 50% de las personas pertenece a un estrato menor o igual a 3 con un rango intercuartílico de 1.25, dandonos a entender que el 50% central de los datos se encuentra entre 1.75 y 3. Por otro lado, el 25% de las reside en un estrato menor o igual a 1.75, y el 75% de las personas en una menor o igual a 3. El estrato mínimo observado fue 1 y el máximo 3.

Ahora, contianuaremos el análisis con las variables catégoricas `Genero` y `Fuma`.

In [3]:
conteo1 = df["Genero"].value_counts()

proporcion1 = df["Genero"].value_counts(normalize=True)

resultado = pd.DataFrame({
    "Genero": conteo1.index,
    "Frecuencia": conteo1.values,
    "Proporción": proporcion1.values * 100
})

resultado

Genero  Frecuencia  Proporción
0  HOMBRE           6        75.0
1   MUJER           2        25.0

De acuerdo al resultado anterior, el 6 de las personas encuestadas son hombres y 2 son mujeres, lo cual corresponde al 75% y 25% respectivamente. De los 8 universitarios encuestados, hay una fuerte predominancia de los hombres. Luego, vayamos con la variable `Fuma`

In [4]:
conteo1 = df["Fuma"].value_counts()

proporcion1 = df["Fuma"].value_counts(normalize=True)

resultado = pd.DataFrame({
    "Fuma": conteo1.index,
    "Frecuencia": conteo1.values,
    "Proporción": proporcion1.values * 100
})

resultado

Fuma  Frecuencia  Proporción
0   NO           5        62.5
1   SI           3        37.5

De acuerdo al resultado anterior, el 5 de las personas encuestadas no fuman y 3 sí fuman, lo cual corresponde al 62.5% y 37.5% respectivamente. De los 8 universitarios encuestados, hay una leve predominancia de las personas que fuman pero no tan fuerte.

3. Continuaremos extrayendo un subconjunto de datos con los primeros 4 estudiantes y lo llamaremos `subc1`.

In [5]:
subc1 = df.iloc[0:4]
subc1

Masa  Estatura  Genero  Estrato Fuma
0    71       183  HOMBRE        2   NO
1    45       168  HOMBRE        1   NO
2    50       164  HOMBRE        1   NO
3    60       164   MUJER        3   SI

4. Seguidamente, extraemos las variables categóricas y las guardamos en un subconjunto llamado `subc2`

In [6]:
subc2 = df.loc[:, ["Genero", "Fuma"]]
subc2

Genero Fuma
0  HOMBRE   NO
1  HOMBRE   NO
2  HOMBRE   NO
3   MUJER   SI
4  HOMBRE   SI
5  HOMBRE   NO
6   MUJER   NO
7  HOMBRE   SI

5. Por último, extraemos las variables numéricas y las guardamos en un subconjunto llamado `subc3`.

In [7]:
subc3 = df.select_dtypes(include="int64")
subc3

Masa  Estatura  Estrato
0    71       183        2
1    45       168        1
2    50       164        1
3    60       164        3
4    61       167        3
5    84       182        3
6    55       168        3
7    60       170        3